lib/calendar.ts

```diff
@@ -445,71 +443,48 @@ impl['iso8601'] = {
     if (!HasSlot(date, ISO_YEAR)) date = ES.ToTemporalDate(date);
     return ES.LeapYear(GetSlot(date, ISO_YEAR));
   }
-};
+} as CalendarImpl;
```

Comment:
Could this instead be something like

```
impl['iso8601'] = new (class implements CalendarImpl {
 .. impl content here ...
 })();
```

which avoids casting and aligns better with the helper registration?

Reply:
Actually, the cast is unnecessary because the `impl` array is typed as `Record<BuiltinCalendarId, CalendarImpl>`. I removed it; see the latest commit.

re: making it a class, the problem is that TS's `class` syntax doesn't automatically type class member functions. So I'd need to add types to every parameter of every method, and after all that work we'd end up with exactly the same level of type safety as the current code. Also, it'd have quite a different runtime emit compared to proposal-temporal.

I've observed while refactoring this file that TS `class` is great when you want to inherit functionality or create multiple instances. But when creating non-inherited singletons (like this object here), it seems like `class` adds work without adding much value because class members don't snap to interface types.

So I'm not sure how best to proceed. Seems like there are these options: 
1) Leave as-is
2) Convert both the ISO and non-ISO `XxxImpl` instances to anonymous classes per your suggestion above, but otherwise leave as-is.
3) Convert `nonIsoHelper` from a cloned singleton into an abstract base class and remove the "helper" concept because each non-ISO calendar would have its own class derived from NonIsoImpl. The original reason for the current split between `XxxImpl` and `helper` was to ensure that the non-ISO implementation was the same shape as the  ISO one. I wanted it to be really obvious when the NonIsoImpl object was calling into a method that didn't exist in ISO. But we don't need to stick with this architecture if we don't think it's helpful.

My preference is for (1) because it's easiest and involves the least deviation from existing proposal-temporal code. But I don't have a very strong opinion either way.  What do you think?

Reply:
> because the `impl` array is typed as `Record<BuiltinCalendarId, CalendarImpl>`. I removed it; see the latest commit.

Actually, I refined the type of the impl array further so that the record is CalendarImpl for the ISO calendar and NonIsoImpl for other calendars. More precise.

Reply:
After thinking about it overnight, this PR is already pretty big so I think the best thing would be to merge this as-is, and then consider (2) or (3) above in a separate PR which would be smaller in scope so would be easier to review. I opened #119 to discuss, and I'll resolve this discussion now.